app/controllers/hosts_controller.rb

```diff
@@ -546,7 +546,8 @@ def process_taxonomy
   end
 
   def template_used
-    host      = Host.new(clean_interfaces_attributes.except(:host_parameters_attributes))
+    host = params[:id] ? Host.find(params[:id]) : Host.new
```

Comment:
It's not caused by `attributes=` but by `puppetclass_ids=`. In this case we don't need this (and other has_many associations) to resolve templates so we can ignore these. The view renders only the templates partial so form is not touched (unlike process_taxonomy action which renders the whole form). I added also config_group_ids which is part of params. To be more safe for associations added in future we could ignore all `/.*_ids$/` keys, not sure if it's any better.